detekt/wrapper/src/main/java/io/buildfoundation/bazel/detekt/ExecutionUtils.java

```diff
@@ -0,0 +1,121 @@
+package io.buildfoundation.bazel.detekt;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class ExecutionUtils {
+    /**
+     * Returns true if run-as-test-target flag is included in arguments
+     */
+    public static boolean shouldRunAsTestTarget(List<String> arguments) {
+        return arguments.contains("--run-as-test-target");
+    }
+
+    /**
+     * Retrieves the output path for the test result from the input arguments.
+     */
+    public static String getExecutionResultOutputPath(List<String> inputArgs) {
+        String outputPath = getArgument(inputArgs, "--execution-result");
+        if (outputPath == null) {
+            System.err.println("File path for execution-result was not set");
+            System.exit(1);
```

Comment:
It's generally a better idea to limit these `System#exit` calls to the main class, otherwise it becomes impossible to test classes like these.

Reply:
+1, and it's easier to follow program control flow when all System exits happen in `main`! Same with `System.err.println`, I'd say put that into `IllegalStateException` message and throw it :)

Reply:
Same with other `System.err.println()` and `System.exit`s below